packages/wallet/src/services/RollbackResubmitter.ts

```diff
@@ -0,0 +1,50 @@
+import { Cardano } from '@cardano-sdk/core';
+import { ConfirmedNewTxAlonzoWithSlot } from './types';
+import { DocumentStore } from '../persistence';
+import { Observable, map, switchMap, tap } from 'rxjs';
+
+// REVIEW: explore how to get the stability window from the network info
+// 3k/f (where k is the security parameter in genesis, and f is the active slot co-efficient parameter
+// in genesis that determines the probability for amount of blocks created in an epoch.)
+const kStabilityWindowSlotsCount = 129_600; // 3k/f
+
+export const createResubmitRollbacks = (
+  rollback$: Observable<Cardano.TxAlonzo[]>,
+  confirmed$: Observable<ConfirmedNewTxAlonzoWithSlot>,
+  store: DocumentStore<ConfirmedNewTxAlonzoWithSlot[]>,
+  stabilityWindowSlotsCount: number = kStabilityWindowSlotsCount
+): Observable<Cardano.NewTxAlonzo[]> => {
+  let confirmedTransactions = [] as ConfirmedNewTxAlonzoWithSlot[];
+  store
+    .get()
+    .pipe(
+      switchMap((transactions) => {
+        // Load stored transactions
+        confirmedTransactions = [...transactions];
+        return confirmed$;
+      })
+    )
+    .subscribe((t) => {
```

Comment:
> Then you won't be able to do [this](https://github.com/input-output-hk/cardano-js-sdk/pull/351/files#diff-fa763eb5d53c8a5623e10eae183f3f7a96bfa37d3762cb856fb324b803906a3fR45), but I think it's fine as we will filter them out anyway after stability window. I'd prefer cleaner implementation over slight optimization.

Actually I realized that this would cause a problem with duplicates (resubmitted and then reconfirmed transactions), so you do need to remove them. Let's keep your implementation of maintaining the array, just maybe refactor it a little bit to do side effects with `tap` and keeping stored transactions in sync with the array:

```ts
confirmedTransactions = confirmedTransactions.filter(({ slot }) => slot > oldestAcceptedSlot);
store.set(confirmedTransactions);
```

Reply:
Added `submitted$` transactions in the logic. See [0286cb7](https://github.com/input-output-hk/cardano-js-sdk/pull/351/commits/0286cb7b5c3844765304acc03f3163345df65fa2)